requirements.txt

```
@@ -4,6 +4,7 @@ Pygments >= 2.9.0
 # See https://github.com/sphinx-doc/sphinx/pull/11100
 Sphinx >= 5.1.1, != 6.1.0, != 6.1.1, < 8.1.0
 docutils >= 0.19.0
+sphinx-notfound-page
```

Seems support for `notfound_urls_prefix = None` [was added in version 1.0.2](https://github.com/readthedocs/sphinx-notfound-page/blob/main/CHANGELOG.rst#version-102)

```suggestion
sphinx-notfound-page >= 1.0.2
```